tests/src/test/java/io/streamnative/pulsar/handlers/kop/compatibility/saslplain/SaslPlainEndToEndTestBase.java

```diff
@@ -0,0 +1,378 @@
+        ((KafkaServiceConfiguration) conf).setKafkaMetadataTenant("internal");
+        ((KafkaServiceConfiguration) conf).setKafkaMetadataNamespace("__kafka");
+
+        conf.setClusterName(super.configClusterName);
+        conf.setAuthorizationEnabled(true);
+        conf.setAuthenticationEnabled(true);
+        conf.setAuthorizationAllowWildcardsMatching(true);
+        conf.setSuperUserRoles(Sets.newHashSet(ADMIN_USER));
+        conf.setAuthenticationProviders(
+                Sets.newHashSet("org.apache.pulsar.broker.authentication."
+                        + "AuthenticationProviderToken"));
+        conf.setBrokerClientAuthenticationPlugin(AuthenticationToken.class.getName());
+        conf.setBrokerClientAuthenticationParameters("token:" + adminToken);
+        conf.setProperties(properties);
+
+        super.internalSetup();
+
+        admin.topics().createPartitionedTopic(TOPIC, 1);
+        admin.namespaces().grantPermissionOnNamespace(TENANT + "/" + NAMESPACE, SIMPLE_USER,
+                Sets.newHashSet(AuthAction.consume, AuthAction.produce));
+    }
+
+    @Override
+    protected void createAdmin() throws Exception {
+        super.admin = spy(PulsarAdmin.builder().serviceHttpUrl(brokerUrl.toString())
+                .authentication(this.conf.getBrokerClientAuthenticationPlugin(),
+                        this.conf.getBrokerClientAuthenticationParameters()).build());
+    }
+
+    @AfterClass
+    protected void cleanup() throws Exception {
+        super.internalCleanup();
+        jaasConfigFile.delete();
+    }
+
+    @AfterMethod
+    protected void cleanJaasConf() {
+        jaasConfigFile.delete();
+        Configuration.setConfiguration(null);
+    }
+
+    protected void writeJaasFile(String password) throws IOException {
+        jaasConfigFile = File.createTempFile("jaas", ".conf");
+        System.setProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM, jaasConfigFile.toString());
+        StringBuilder builder = new StringBuilder();
+        builder.append("\torg.apache.kafka.common.security.plain.PlainLoginModule");
+        builder.append(' ');
+        builder.append("required \n");
+        builder.append('\t');
+        builder.append("username");
+        builder.append("=\"");
+        builder.append("public/default\" \n");
+        builder.append('\t');
+        builder.append("password");
+        builder.append("=\"");
+        builder.append(password);
+        builder.append("\";");
+        List<String> lines = Arrays.asList("KafkaClient {", builder.toString(), "};");
+        Files.write(jaasConfigFile.toPath(), lines, StandardCharsets.UTF_8);
+        System.out.println(jaasConfigFile.getPath());
+    }
+
+    @Test(timeOut = 120000)
+    void testKafkaProduceAndConsumeWithSaslPlain() throws Exception {
+
+        // need java.security.auth.login.config for sasl/plain in kafka client 0.10.0.0
+        writeJaasFile("token:" + userToken);
```

Comment:
Kafka client version 0.10.0.0, sasl/plain authentication needs to be configured `java.security.auth.login.config`